--- models.py	2014-09-21 21:52:45.931951688 +0200
+++ fixmodels.py	2014-09-21 21:53:43.431951869 +0200
@@ -551,9 +551,7 @@
         return super(BaseModelFormSet, self).initial_form_count()
 
     def _existing_object(self, pk):
-        if not hasattr(self, '_object_dict'):
-            self._object_dict = dict([(o.pk, o) for o in self.get_queryset()])
-        return self._object_dict.get(pk)
+        return self.model._default_manager.get_query_set().get(pk = pk)
 
     def _construct_form(self, i, **kwargs):
         if self.is_bound and i < self.initial_form_count():
